src/backend/distributed/commands/foreign_constraint.c

```diff
@@ -33,68 +35,45 @@
 #include "utils/ruleutils.h"
 #include "utils/syscache.h"
 
+/*
+ * Flags that can be passed to GetTableForeignConstraintOidsInternal to indicate
+ * which foreign key constraint OIDs are to be extracted
+ */
+#define EXTRACT_REFERENCING_CONSTRAINTS 0x0001
```

Comment:
> Here it would be nice to use bits instead of "hexa", currently the values are 1 << 0 (0x0001), 1 << 4 (0x00010). But they could be 1 << 0, 1 << 1, 1 << 2 etc(the important part is that they shouldn't have conflicting bits) and capturing this in an enum could be good too, an example:

Using enum and bit-shift representation makes sense to me.

But I'm not sure about using a `find` on the contrary of an `exclude` one for self references. This is because, I think the default should be "including the self references" too.

Reply:
Enum is also useful to enforce passing accurate flags

Reply:
I should note that it might still make sense to pass `int` (or uint32) as a flag since combination of the flags would not be a member of the enum

Reply:
I think EXTRACT could mean "exclude" as well but here we use it as "include" so what about using "INCLUDE" and "EXCLUDE" prefixes?